opal/corelib/kernel.rb

```diff
@@ -504,6 +504,12 @@ def Float(value)
 
       return #{::Opal.coerce_to!(value, ::Float, :to_f)};
     }
+  rescue
+    unless exception
+      return nil
+    end
+
+    ::Kernel.raise
```

Comment:
Thoughts on using the `$coerce_to` helper instead of rescuing whenever `exception:` is false?

Something like:

```rb
%x{
  // …
  return (#{eception} ? #{::Opal.coerce_to!(value, ::Float, :to_f)} : $coerce_to(value, #{::Float}, 'to_f')
}
```

Reply:
🤔 There's also the `Kernel.raise` in the inner `if`… that would mean nesting an `if (!exception) return nil` there.

Please let me know your thoughts, the `rescue` complicates the compiled code and worsen performance significantly, but I also see a point in using a simpler implementation or ruby's side.

Reply:
`$coerce_to` brings no too complicated code for us, I think.

I used `rescue` because  I didn't now `$coerce_to`. I'll replace `rescue` with `$coerce_to`.